src/campaign/components/ItemCard/__test__/TagGroup.test.tsx

```diff
@@ -0,0 +1,50 @@
+import userEvent from "@testing-library/user-event";
+import { render, screen } from "shared";
+import TagGroup from "../TagGroup";
+
+describe("TagGroup", () => {
+  const tags = [
+    "tag-1",
+    "tag-2",
+    "tag-3",
+    "tag-4",
+    "tag-5",
+    "tag-6",
+    "tag-7",
+    "tag-8",
+  ];
+  it("should display provided tags when no limit is set", () => {
+    render(<TagGroup tags={tags} />);
+    tags.forEach((tag) => expect(screen.getByText(tag)).toBeVisible());
+  });
+
+  it("should display allowed number only when display limit is set", () => {
+    const limit = 3;
+    render(<TagGroup tags={tags} displayLimit={limit} />);
+    tags.forEach((tag, index) => {
+      index < limit
+        ? expect(screen.getByText(tag)).toBeVisible()
+        : expect(screen.queryByText(tag)).not.toBeInTheDocument();
+    });
+  });
+
+  describe('"show all tags" button', () => {
+    it("should be visible when limit is less than tag count", () => {
+      const limit = 3;
+      render(<TagGroup tags={tags} displayLimit={limit} />);
```

Comment:
Can we split up tests into 3 blocks, makes them easier to read. 

Setup

Run

Assert